# Preparing data for uploading to database
### I use pandas for easy manipulation

In [74]:
import pandas as pd
import numpy as np
import datetime


# Load csv into Dataframe
filename = 'epg.csv'
df = pd.read_csv(filename)
df.head(1)

channel_id  channel_name channel_country  start_date start_time  \
0   384303590  HD1 (France)             FRA    20180101      00:00   

   duration_in_seconds  program_id program_original_title program_local_title  \
0                84000      461044          À communiquer       À communiquer   

   program_year  
0           NaN

In [75]:
channel_ids = df['channel_id'].unique()
# [384303590,  69036190,  69036687]
channel_names = df['channel_name'].unique()
# ['HD1 (France)', 'La Sexta', 'RTE 1']
channel_countries = df['channel_country'].unique() 
# ['FRA', 'ESP', 'IRL']
program_ids = df['program_id'].unique().size
# 1879
program_original_title = df['program_original_title'].unique().size
# 358
program_local_title = df['program_local_title'].unique().size
# 358

# What is the shape of the df
print("The data is {} cols by {} rows.".format(df.shape[1], df.shape[0]))
for column in df.columns:
    print(column, df[column].count())

print('-'*31)  
print('I can tell there are cells in "program_original_title" and "program_year" that are empty or NaN')

print('The quantity of unique values in each col are the following:')
for column in df.columns:
    print(column, df[column].unique().size)

The data is 10 cols by 4897 rows.
('channel_id', 4897)
('channel_name', 4897)
('channel_country', 4897)
('start_date', 4897)
('start_time', 4897)
('duration_in_seconds', 4897)
('program_id', 4897)
('program_original_title', 4860)
('program_local_title', 4897)
('program_year', 2464)
-------------------------------
I can tell there are cells in "program_original_title" and "program_year" that are empty or NaN
The quantity of unique values in each col are the following:
('channel_id', 3)
('channel_name', 3)
('channel_country', 3)
('start_date', 59)
('start_time', 266)
('duration_in_seconds', 98)
('program_id', 1879)
('program_original_title', 358)
('program_local_title', 372)
('program_year', 44)


### Fill null titles and years with n/a. 
```python
df = df.fillna('n/a)
```

In [76]:
df = df.fillna('n/a')
df.dtypes

channel_id                 int64
channel_name              object
channel_country           object
start_date                 int64
start_time                object
duration_in_seconds        int64
program_id                 int64
program_original_title    object
program_local_title       object
program_year              object
dtype: object

In [77]:
# Change DataFrame dtypes to match Database models
def changeDtypesForDb(df):
    df['program_year'] = pd.to_numeric(df['program_year'], errors='coerce')
    df['start_date'] = pd.to_datetime(df['start_date'], format='%Y%m%d', errors='ignore')
    df['start_time'] = df['start_time'].map(lambda x: datetime.datetime.strptime(x, '%H:%M').time())
    df['duration_in_seconds'] = pd.to_timedelta(df['duration_in_seconds'], unit='seconds')
    return df

In [78]:
# Add the df Showtimes to DB
df = changeDtypesForDb(df)
df.dtypes

channel_id                          int64
channel_name                       object
channel_country                    object
start_date                 datetime64[ns]
start_time                         object
duration_in_seconds       timedelta64[ns]
program_id                          int64
program_original_title             object
program_local_title                object
program_year                      float64
dtype: object

In [79]:
from tv import models
for object in models.Program.objects.all():
    object.delete()
for object in models.Channel.objects.all():
    object.delete()
for object in models.Showtime.objects.all():
    object.delete()

In [80]:
from django.db import IntegrityError
from tv import models


# Create Program with channel obj and row data
def createProgram(row, channel):
    try:
        program = models.Program(uid=row.program_id, 
                                 original_title=row.program_original_title,
                                 local_title=row.program_local_title, 
                                 year=row.program_year,
                                 channel=channel)
        program.save()
        print("Unique Program {} with id {} added to DB".format(program.local_title, program.uid))
        return program
    except IntegrityError:
        print("Program with unique id {} already exists".format(program.uid))
        return
    

# Create Showtime with program obj and row data
def createShowtime(row, program):
    try:
        showtime = models.Showtime(start_date=row.start_date, start_time=row.start_time,
                                           duration=row.duration_in_seconds, program=program)
        showtime.save()
        print('Showtime: {} on {} was added to DB.'.format(showtime.start_date,program.local_title))
        return showtime
    except IntegrityError:
        print("Showtime already exists in DB".format(showtime.pk))
        return
    
              
# Create Showtimes from DataFrame           
def addShowtimeFromDataFrame(df):
    for index, row in df.iterrows():
        channel, create = models.Channel.objects.get_or_create(uid=row.channel_id, 
                                                               name=row.channel_name,
                                                               country_code=row.channel_country)
        if not create:
            program = createProgram(row, channel)
            showtime = createShowtime(row, program)
        else:
            channel.name = row.channel_name
            channel.country_code = row.channel_country
            channel.save()
            program = createProgram(row, channel)
            showtime = createShowtime(row, program)
    print("Finished adding programs from {}".format(filename))
    return showtime
        


# Add to DB
```python
addShowtimeFromDataFrame(df)
```

In [81]:
addShowtimeFromDataFrame(df)
df.isnull()

Unique Program À communiquer with id 461044 added to DB
Showtime: 2018-01-01 00:00:00 on À communiquer was added to DB.
Unique Program Será anunciado with id 1887442 added to DB
Showtime already exists in DB
Unique Program RTE 1 Programming with id 3452333967 added to DB
Showtime already exists in DB
Unique Program L.A. Confidential with id 446381 added to DB
Showtime: 2018-01-01 00:00:00 on L.A. Confidential was added to DB.
Unique Program No controles with id 18063328 added to DB
Showtime: 2018-01-01 00:00:00 on No controles was added to DB.
Unique Program Canal Ruleta with id 5072179538 added to DB
Showtime: 2018-01-01 00:00:00 on Canal Ruleta was added to DB.
Unique Program Minutos musicales with id 11742116 added to DB
Showtime: 2018-01-01 00:00:00 on Minutos musicales was added to DB.
Unique Program Papillon with id 1064550 added to DB
Showtime already exists in DB
Program with unique id 11742116 already exists
Showtime already exists in DB
Unique Program Bestial with id 19219963

Unique Program Las primeras 48 horas with id 4804635922 added to DB
Showtime already exists in DB
Unique Program Au nom de la vérité with id 5601168087 added to DB
Showtime: 2018-01-02 00:00:00 on Au nom de la vérité was added to DB.
Unique Program Al Rojo Vivo with id 5668214430 added to DB
Showtime: 2018-01-02 00:00:00 on Al Rojo Vivo was added to DB.
Unique Program Au nom de la vérité with id 5542565692 added to DB
Showtime: 2018-01-02 00:00:00 on Au nom de la vérité was added to DB.
Unique Program Au nom de la vérité with id 5569923357 added to DB
Showtime: 2018-01-02 00:00:00 on Au nom de la vérité was added to DB.
Unique Program Au nom de la vérité with id 1156185999 added to DB
Showtime: 2018-01-02 00:00:00 on Au nom de la vérité was added to DB.
Unique Program Shortland Street with id 5612079579 added to DB
Showtime: 2018-01-02 00:00:00 on Shortland Street was added to DB.
Unique Program Au nom de la vérité with id 5547041841 added to DB
Showtime: 2018-01-02 00:00:00 on Au nom 

Program with unique id 17875416 already exists
Showtime already exists in DB
Program with unique id 12766064 already exists
Showtime already exists in DB
Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Program with unique id 17875409 already exists
Showtime already exists in DB
Unique Program Reeling in the Years with id 12068169 added to DB
Showtime: 2018-01-03 00:00:00 on Reeling in the Years was added to DB.
Unique Program Grey's Anatomy with id 12887841 added to DB
Showtime: 2018-01-03 00:00:00 on Grey's Anatomy was added to DB.
Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program Nationwide with id 5612079557 added to DB
Showtime already exists in DB
Unique Program Neven's Food Trails: The Basque Country with id 4089228914 added to DB
Showtime: 2018-01-03 00:00:00 on Neven's Food Trails: The Basque Country was added to DB.
Program with u

Showtime: 2018-01-04 00:00:00 on EastEnders was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program Fair City with id 5612079573 added to DB
Showtime: 2018-01-04 00:00:00 on Fair City was added to DB.
Unique Program V pour Vendetta with id 2039561 added to DB
Showtime already exists in DB
Program with unique id 11922093 already exists
Showtime already exists in DB
Unique Program Ear to the Ground with id 5612079537 added to DB
Showtime: 2018-01-04 00:00:00 on Ear to the Ground was added to DB.
Unique Program El intermedio Christmas Edition with id 5668214454 added to DB
Showtime already exists in DB
Program with unique id 12147320 already exists
Showtime already exists in DB
Unique Program Dos madres perfectas with id 666254531 added to DB
Showtime: 2018-01-04 00:00:00 on Dos madres perfectas was added to DB.
Unique Program Daddy's Home with id 863726785 a

Showtime: 2018-01-05 00:00:00 on Scannal was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program Pat Shortt's Music From D'Telly with id 1982269411 added to DB
Showtime: 2018-01-05 00:00:00 on Pat Shortt's Music From D'Telly was added to DB.
Unique Program Nos chers voisins with id 4442942482 added to DB
Showtime already exists in DB
Program with unique id 11922093 already exists
Showtime already exists in DB
Unique Program Francis Brennan's Grand Vietnamese Tour with id 5033379015 added to DB
Showtime: 2018-01-05 00:00:00 on Francis Brennan's Grand Vietnamese Tour was added to DB.
Unique Program #laSexta Columna with id 22365312 added to DB
Showtime already exists in DB
Program with unique id 12147320 already exists
Showtime already exists in DB
Unique Program Equipo de investigación with id 21714748 added to DB
Showtime: 2018-01-05 00:00:00 on Equipo de 

Unique Program Senior Moments with id 5656763574 added to DB
Showtime: 2018-01-06 00:00:00 on Senior Moments was added to DB.
Unique Program laSexta Noche with id 5668214458 added to DB
Showtime: 2018-01-06 00:00:00 on laSexta Noche was added to DB.
Program with unique id 12147320 already exists
Showtime already exists in DB
Unique Program The Ray D'Arcy Show with id 1046903433 added to DB
Showtime: 2018-01-06 00:00:00 on The Ray D'Arcy Show was added to DB.
Unique Program Une mission pour Noël with id 4273628665 added to DB
Showtime: 2018-01-06 00:00:00 on Une mission pour Noël was added to DB.
Unique Program The Nathan Carter Show with id 4175358936 added to DB
Showtime: 2018-01-06 00:00:00 on The Nathan Carter Show was added to DB.
Program with unique id 13358395 already exists
Showtime already exists in DB
Unique Program Parkland with id 982759810 added to DB
Showtime: 2018-01-06 00:00:00 on Parkland was added to DB.
Program with unique id 1887442 already exists
Showtime already ex

Unique Program Crímenes imperfectos with id 11817363 added to DB
Showtime: 2018-01-08 00:00:00 on Crímenes imperfectos was added to DB.
Unique Program Kill Zone USA with id 4089232280 added to DB
Showtime: 2018-01-08 00:00:00 on Kill Zone USA was added to DB.
Program with unique id 1076761458 already exists
Showtime already exists in DB
Unique Program Call the Midwife with id 22350147 added to DB
Showtime: 2018-01-08 00:00:00 on Call the Midwife was added to DB.
Program with unique id 5072179538 already exists
Showtime already exists in DB
Program with unique id 11742116 already exists
Showtime already exists in DB
Unique Program Call the Midwife with id 22387167 added to DB
Showtime: 2018-01-08 00:00:00 on Call the Midwife was added to DB.
Unique Program The Nanny with id 439645 added to DB
Showtime: 2018-01-08 00:00:00 on The Nanny was added to DB.
Unique Program The Nanny with id 439643 added to DB
Showtime: 2018-01-08 00:00:00 on The Nanny was added to DB.
Program with unique id 12

Unique Program Neighbours with id 5644925644 added to DB
Showtime: 2018-01-09 00:00:00 on Neighbours was added to DB.
Program with unique id 3286898281 already exists
Showtime already exists in DB
Program with unique id 5641846299 already exists
Showtime already exists in DB
Program with unique id 768381956 already exists
Showtime already exists in DB
Program with unique id 12744336 already exists
Showtime already exists in DB
Unique Program Zapeando with id 5685798395 added to DB
Showtime: 2018-01-09 00:00:00 on Zapeando was added to DB.
Unique Program How to Cook Well, With Rory O'Connell with id 3509191794 added to DB
Showtime: 2018-01-09 00:00:00 on How to Cook Well, With Rory O'Connell was added to DB.
Program with unique id 4113170753 already exists
Showtime already exists in DB
Unique Program Urgences with id 12571025 added to DB
Showtime: 2018-01-09 00:00:00 on Urgences was added to DB.
Unique Program Más vale tarde with id 5685798399 added to DB
Showtime: 2018-01-09 00:00:00 o

Showtime: 2018-01-10 00:00:00 on Zapeando was added to DB.
Program with unique id 5656763521 already exists
Showtime already exists in DB
Program with unique id 4113170753 already exists
Showtime already exists in DB
Unique Program Urgences with id 13067250 added to DB
Showtime: 2018-01-10 00:00:00 on Urgences was added to DB.
Unique Program Más vale tarde with id 5685798409 added to DB
Showtime: 2018-01-10 00:00:00 on Más vale tarde was added to DB.
Program with unique id 12718811 already exists
Showtime already exists in DB
Program with unique id 12740306 already exists
Showtime already exists in DB
Program with unique id 12766064 already exists
Showtime already exists in DB
Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Program with unique id 12930014 already exists
Showtime already exists in DB
Unique Program Grey's Anatomy with id 12518853 added to DB
Showtime: 2018-01-10 00:

Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program Grey's Anatomy with id 13318909 added to DB
Showtime: 2018-01-11 00:00:00 on Grey's Anatomy was added to DB.
Unique Program Getaways with id 1376446508 added to DB
Showtime already exists in DB
Unique Program EastEnders with id 5641846301 added to DB
Showtime: 2018-01-11 00:00:00 on EastEnders was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program Fair City with id 5656763523 added to DB
Showtime: 2018-01-11 00:00:00 on Fair City was added to DB.
Unique Program Nikita with id 12019208 added to DB
Showtime already exists in DB
Program with unique id 11922093 already exists
Showtime already exists in DB
Unique Program El intermedio with id 5685798385 added to DB
Showtime: 2018-01-11 00:00:00 on El intermedio was added to DB.
Unique Program Ear to the Ground with id 

Unique Program EastEnders with id 5641846297 added to DB
Showtime: 2018-01-12 00:00:00 on EastEnders was added to DB.
Unique Program Les innocents with id 5652181298 added to DB
Showtime already exists in DB
Program with unique id 11922093 already exists
Showtime already exists in DB
Program with unique id 22365312 already exists
Showtime already exists in DB
Program with unique id 5033379015 already exists
Showtime already exists in DB
Unique Program Les innocents with id 5652187214 added to DB
Showtime: 2018-01-12 00:00:00 on Les innocents was added to DB.
Program with unique id 12147320 already exists
Showtime already exists in DB
Program with unique id 21714748 already exists
Showtime already exists in DB
Unique Program The Late Late Show with id 5713849580 added to DB
Showtime: 2018-01-12 00:00:00 on The Late Late Show was added to DB.
Program with unique id 12116683 already exists
Showtime already exists in DB
Unique Program Equipo de investigación with id 5395769834 added to DB


Unique Program Life in Pieces with id 3358434894 added to DB
Showtime: 2018-01-14 00:00:00 on Life in Pieces was added to DB.
Unique Program Our Girl with id 1211195994 added to DB
Showtime: 2018-01-14 00:00:00 on Our Girl was added to DB.
Unique Program Waking the Dead with id 8727006 added to DB
Showtime: 2018-01-14 00:00:00 on Waking the Dead was added to DB.
Program with unique id 11742116 already exists
Showtime already exists in DB
Unique Program Historias con denominación de origen with id 1416799300 added to DB
Showtime: 2018-01-14 00:00:00 on Historias con denominación de origen was added to DB.
Program with unique id 12073476 already exists
Showtime already exists in DB
Program with unique id 4074450578 already exists
Showtime already exists in DB
Unique Program Bestial with id 19059452 added to DB
Showtime: 2018-01-14 00:00:00 on Bestial was added to DB.
Program with unique id 4074450416 already exists
Showtime already exists in DB
Program with unique id 12185596 already exi

Program with unique id 5578795126 already exists
Showtime already exists in DB
Program with unique id 4113170753 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 17719433 added to DB
Showtime: 2018-01-15 00:00:00 on Crímenes imperfectos was added to DB.
Program with unique id 5654395875 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 17719434 added to DB
Showtime: 2018-01-15 00:00:00 on Crímenes imperfectos was added to DB.
Unique Program Las primeras 48 horas with id 67223398 added to DB
Showtime: 2018-01-15 00:00:00 on Las primeras 48 horas was added to DB.
Program with unique id 5675873798 already exists
Showtime already exists in DB
Program with unique id 5632268723 already exists
Showtime already exists in DB
Unique Program Al Rojo Vivo with id 5696564851 added to DB
Showtime: 2018-01-15 00:00:00 on Al Rojo Vivo was added to DB.
Unique Program The Doctors with id 4544341876 added to DB
Showtime: 2018-

Program with unique id 5620807915 already exists
Showtime already exists in DB
Unique Program The Shelbourne with id 1148922141 added to DB
Showtime: 2018-01-16 00:00:00 on The Shelbourne was added to DB.
Unique Program Au nom de la vérité with id 5685576509 added to DB
Showtime: 2018-01-16 00:00:00 on Au nom de la vérité was added to DB.
Program with unique id 12127388 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 13038226 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5644925638 added to DB
Showtime: 2018-01-16 00:00:00 on Home And Away was added to DB.
Program with unique id 13122042 already exists
Showtime already exists in DB
Program with unique id 4077761636 already exists
Showtime already exists 

Unique Program Urgences with id 18796711 added to DB
Showtime: 2018-01-17 00:00:00 on Urgences was added to DB.
Unique Program Más vale tarde with id 5702936077 added to DB
Showtime: 2018-01-17 00:00:00 on Más vale tarde was added to DB.
Program with unique id 13318913 already exists
Showtime already exists in DB
Program with unique id 13648962 already exists
Showtime already exists in DB
Program with unique id 12766064 already exists
Showtime already exists in DB
Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Program with unique id 17593682 already exists
Showtime already exists in DB
Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program Nationwide with id 5721295902 added to DB
Showtime: 2018-01-17 00:00:00 on Nationwide was added to DB.
Unique Program Grey's Anatomy with id 17593683 added to DB
Showtime already exists in DB
Unique Program 

Showtime already exists in DB
Program with unique id 17593682 already exists
Showtime already exists in DB
Program with unique id 12766064 already exists
Showtime already exists in DB
Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Program with unique id 17593683 already exists
Showtime already exists in DB
Unique Program Grey's Anatomy with id 13721132 added to DB
Showtime: 2018-01-18 00:00:00 on Grey's Anatomy was added to DB.
Unique Program Getaways with id 1384631800 added to DB
Showtime already exists in DB
Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program EastEnders with id 5698558894 added to DB
Showtime: 2018-01-18 00:00:00 on EastEnders was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program Léon with

Unique Program Fleadh Cheoil with id 5739068009 added to DB
Showtime: 2018-01-19 00:00:00 on Fleadh Cheoil was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program EastEnders with id 5698558892 added to DB
Showtime: 2018-01-19 00:00:00 on EastEnders was added to DB.
Unique Program Les innocents with id 5671138125 added to DB
Showtime already exists in DB
Program with unique id 11922093 already exists
Showtime already exists in DB
Unique Program Francis Brennan's Grand Vietnamese Tour with id 5060385023 added to DB
Showtime: 2018-01-19 00:00:00 on Francis Brennan's Grand Vietnamese Tour was added to DB.
Program with unique id 22365312 already exists
Showtime already exists in DB
Unique Program Les innocents with id 5671138155 added to DB
Showtime: 2018-01-19 00:00:00 on Les innocents was added to DB.
Program with unique id 12147320 already exists
Showtime al

Unique Program laSexta Noche with id 5717843676 added to DB
Showtime: 2018-01-20 00:00:00 on laSexta Noche was added to DB.
Program with unique id 12147320 already exists
Showtime already exists in DB
Unique Program Au-dessus de tout soupçon - 2 with id 5709170360 added to DB
Showtime: 2018-01-20 00:00:00 on Au-dessus de tout soupçon - 2 was added to DB.
Unique Program The Ray D'Arcy Show with id 5717709415 added to DB
Showtime: 2018-01-20 00:00:00 on The Ray D'Arcy Show was added to DB.
Program with unique id 5698201851 already exists
Showtime already exists in DB
Program with unique id 13358395 already exists
Showtime already exists in DB
Program with unique id 1887442 already exists
Showtime already exists in DB
Program with unique id 3452333967 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Unique Program Field Punishment No 1 with id 1215842008 added to DB
Showtime: 2018-01-21 00:00:00 on Field Punishment No 

Program with unique id 11742116 already exists
Showtime already exists in DB
Program with unique id 1849903043 already exists
Showtime already exists in DB
Program with unique id 12073476 already exists
Showtime already exists in DB
Program with unique id 1849903043 already exists
Showtime already exists in DB
Program with unique id 14270681 already exists
Showtime already exists in DB
Program with unique id 12185596 already exists
Showtime already exists in DB
Program with unique id 1156186040 already exists
Showtime already exists in DB
Program with unique id 1147198864 already exists
Showtime already exists in DB
Unique Program Petits secrets entre voisins with id 1156186035 added to DB
Showtime: 2018-01-22 00:00:00 on Petits secrets entre voisins was added to DB.
Program with unique id 4113170753 already exists
Showtime already exists in DB
Program with unique id 5642731384 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 17853168 added to DB

Showtime already exists in DB
Unique Program Petits secrets entre voisins with id 1156186044 added to DB
Showtime: 2018-01-23 00:00:00 on Petits secrets entre voisins was added to DB.
Unique Program Petits secrets entre voisins with id 1441660885 added to DB
Showtime: 2018-01-23 00:00:00 on Petits secrets entre voisins was added to DB.
Program with unique id 4113170753 already exists
Showtime already exists in DB
Program with unique id 5617776828 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 17865540 added to DB
Showtime: 2018-01-23 00:00:00 on Crímenes imperfectos was added to DB.
Program with unique id 5617775794 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 17868768 added to DB
Showtime: 2018-01-23 00:00:00 on Crímenes imperfectos was added to DB.
Program with unique id 5702487651 already exists
Showtime already exists in DB
Unique Program Las primeras 48 horas with id 22325073 added to DB
Showtime

Unique Program Dr. Phil with id 4502396251 added to DB
Showtime: 2018-01-24 00:00:00 on Dr. Phil was added to DB.
Program with unique id 5696101457 already exists
Showtime already exists in DB
Unique Program Shortland Street with id 5717709403 added to DB
Showtime: 2018-01-24 00:00:00 on Shortland Street was added to DB.
Program with unique id 5691609508 already exists
Showtime already exists in DB
Program with unique id 5675873798 already exists
Showtime already exists in DB
Program with unique id 11742368 already exists
Showtime already exists in DB
Unique Program Au nom de la vérité with id 5710783827 added to DB
Showtime: 2018-01-24 00:00:00 on Au nom de la vérité was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 13125766 already exists
Showtime already exists in D

Program with unique id 3286898281 already exists
Showtime already exists in DB
Unique Program Catherine's Family Kitchen with id 22485710 added to DB
Showtime: 2018-01-25 00:00:00 on Catherine's Family Kitchen was added to DB.
Program with unique id 768381956 already exists
Showtime already exists in DB
Program with unique id 12929978 already exists
Showtime already exists in DB
Program with unique id 828526801 already exists
Showtime already exists in DB
Program with unique id 5717709357 already exists
Showtime already exists in DB
Program with unique id 4113170753 already exists
Showtime already exists in DB
Unique Program Urgences with id 13170398 added to DB
Showtime: 2018-01-25 00:00:00 on Urgences was added to DB.
Unique Program Más vale tarde with id 328017822 added to DB
Showtime: 2018-01-25 00:00:00 on Más vale tarde was added to DB.
Program with unique id 18048002 already exists
Showtime already exists in DB
Program with unique id 18082399 already exists
Showtime already exis

Showtime already exists in DB
Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Program with unique id 17640623 already exists
Showtime already exists in DB
Unique Program Grey's Anatomy with id 18133293 added to DB
Showtime: 2018-01-26 00:00:00 on Grey's Anatomy was added to DB.
Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program Nationwide with id 5717709373 added to DB
Showtime already exists in DB
Unique Program Fleadh Cheoil with id 5717709355 added to DB
Showtime: 2018-01-26 00:00:00 on Fleadh Cheoil was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program EastEnders with id 5720688485 added to DB
Showtime: 2018-01-26 00:00:00 on EastEnders was added to DB.
Unique Program Les innocents with id 5691352985 adde

Showtime: 2018-01-27 00:00:00 on Training Day was added to DB.
Program with unique id 12108469 already exists
Showtime already exists in DB
Unique Program Training Day with id 5388646549 added to DB
Showtime: 2018-01-27 00:00:00 on Training Day was added to DB.
Unique Program The Ray D'Arcy Show with id 5741755358 added to DB
Showtime: 2018-01-27 00:00:00 on The Ray D'Arcy Show was added to DB.
Unique Program The Walking Dead with id 1337134576 added to DB
Showtime: 2018-01-27 00:00:00 on The Walking Dead was added to DB.
Program with unique id 5717709472 already exists
Showtime already exists in DB
Unique Program The Walking Dead with id 1337134578 added to DB
Showtime: 2018-01-27 00:00:00 on The Walking Dead was added to DB.
Program with unique id 1887442 already exists
Showtime already exists in DB
Program with unique id 3452333967 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Unique Program The Taking of Pelh

Showtime: 2018-01-29 00:00:00 on Al Rojo Vivo was added to DB.
Unique Program The Doctors with id 4566022240 added to DB
Showtime: 2018-01-29 00:00:00 on The Doctors was added to DB.
Program with unique id 5617775794 already exists
Showtime already exists in DB
Program with unique id 5710783827 already exists
Showtime already exists in DB
Unique Program Dr. Phil with id 4502409309 added to DB
Showtime: 2018-01-29 00:00:00 on Dr. Phil was added to DB.
Program with unique id 5654395875 already exists
Showtime already exists in DB
Program with unique id 5696101457 already exists
Showtime already exists in DB
Unique Program Shortland Street with id 5741755342 added to DB
Showtime: 2018-01-29 00:00:00 on Shortland Street was added to DB.
Unique Program Au nom de la vérité with id 5726138304 added to DB
Showtime: 2018-01-29 00:00:00 on Au nom de la vérité was added to DB.
Unique Program The Shelbourne with id 1992647441 added to DB
Showtime: 2018-01-29 00:00:00 on The Shelbourne was added to

Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 13170398 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5721194245 added to DB
Showtime: 2018-01-30 00:00:00 on Home And Away was added to DB.
Program with unique id 13019849 already exists
Showtime already exists in DB
Program with unique id 4077761636 already exists
Showtime already exists in DB
Unique Program Neighbours with id 5728773120 added to DB
Showtime: 2018-01-30 00:00:00 on Neighbours was added to DB.
Program with unique id 3286898281 already exists
Showtime already exists in DB
Program with unique id 5741989637 already exists
Showtime already exists in DB
Program with unique id 768381956 already exists
Showtime already exists in DB
Unique Program Zapeando with id 5748722958 added to DB
Sho

Showtime: 2018-02-01 00:00:00 on Vogue Williams - On the Edge was added to DB.
Program with unique id 505649787 already exists
Showtime already exists in DB
Unique Program It's a Park's Life with id 5207549628 added to DB
Showtime: 2018-02-01 00:00:00 on It's a Park's Life was added to DB.
Program with unique id 1076761458 already exists
Showtime already exists in DB
Unique Program Frock Finders with id 5341030640 added to DB
Showtime: 2018-02-01 00:00:00 on Frock Finders was added to DB.
Unique Program Find Me a Home with id 4066715466 added to DB
Showtime: 2018-02-01 00:00:00 on Find Me a Home was added to DB.
Program with unique id 11742116 already exists
Showtime already exists in DB
Unique Program Celebrity Operation Transformation with id 4036186478 added to DB
Showtime: 2018-02-01 00:00:00 on Celebrity Operation Transformation was added to DB.
Unique Program Home Of The Year with id 4641223804 added to DB
Showtime: 2018-02-01 00:00:00 on Home Of The Year was added to DB.
Unique 

Program with unique id 12127388 already exists
Showtime already exists in DB
Program with unique id 5741755346 already exists
Showtime already exists in DB
Program with unique id 4522798692 already exists
Showtime already exists in DB
Program with unique id 11742116 already exists
Showtime already exists in DB
Program with unique id 12073476 already exists
Showtime already exists in DB
Program with unique id 1878663043 already exists
Showtime already exists in DB
Program with unique id 12073476 already exists
Showtime already exists in DB
Program with unique id 1209380560 already exists
Showtime already exists in DB
Program with unique id 12185596 already exists
Showtime already exists in DB
Program with unique id 14270681 already exists
Showtime already exists in DB
Program with unique id 1209380556 already exists
Showtime already exists in DB
Program with unique id 1147198869 already exists
Showtime already exists in DB
Program with unique id 1329062 already exists
Showtime already e

Program with unique id 4089226957 already exists
Showtime already exists in DB
Program with unique id 4074450300 already exists
Showtime already exists in DB
Unique Program Piso compartido with id 1464017984 added to DB
Showtime: 2018-02-03 00:00:00 on Piso compartido was added to DB.
Unique Program WPC 56 with id 1444538475 added to DB
Showtime: 2018-02-03 00:00:00 on WPC 56 was added to DB.
Program with unique id 4081732901 already exists
Showtime already exists in DB
Program with unique id 4074450392 already exists
Showtime already exists in DB
Unique Program Hoteles con encanto with id 5539603315 added to DB
Showtime: 2018-02-03 00:00:00 on Hoteles con encanto was added to DB.
Unique Program Only Fools and Horses with id 349827 added to DB
Showtime: 2018-02-03 00:00:00 on Only Fools and Horses was added to DB.
Program with unique id 4085282947 already exists
Showtime already exists in DB
Program with unique id 5740316049 already exists
Showtime already exists in DB
Program with uni

Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 5741755284 already exists
Showtime already exists in DB
Program with unique id 2047977 already exists
Showtime already exists in DB
Program with unique id 11955265 already exists
Showtime already exists in DB
Program with unique id 11922093 already exists
Showtime already exists in DB
Program with unique id 768381956 already exists
Showtime already exists in DB
Program with unique id 2063570 already exists
Showtime already exists in DB
Unique Program No hay salida with id 13131685 added to DB
Showtime: 2018-02-04 00:00:00 on No hay salida was added to DB.
Unique Program Earth's Great Seasons with id 5758414357 added to DB
Showtime: 2018-02-04 00:00:00 on Earth's Great Seasons was added to DB.
Program with unique id 2071039 already exists
Showtime already exists in DB
Unique Program 

Showtime: 2018-02-05 00:00:00 on Stetsons And Stilettos was added to DB.
Unique Program El intermedio with id 5764962055 added to DB
Showtime already exists in DB
Program with unique id 12147320 already exists
Showtime already exists in DB
Program with unique id 11982494 already exists
Showtime already exists in DB
Unique Program We Won the Lotto with id 5774958374 added to DB
Showtime: 2018-02-05 00:00:00 on We Won the Lotto was added to DB.
Program with unique id 1163254062 already exists
Showtime already exists in DB
Unique Program Claire Byrne Live with id 5774935210 added to DB
Showtime: 2018-02-05 00:00:00 on Claire Byrne Live was added to DB.
Program with unique id 11982494 already exists
Showtime already exists in DB
Program with unique id 5741755292 already exists
Showtime already exists in DB
Program with unique id 13358395 already exists
Showtime already exists in DB
Unique Program Top Dog, el perro sargento with id 12991801 added to DB
Showtime: 2018-02-05 00:00:00 on Top D

Program with unique id 4544388685 already exists
Showtime already exists in DB
Program with unique id 11742116 already exists
Showtime already exists in DB
Program with unique id 12073476 already exists
Showtime already exists in DB
Program with unique id 1434944092 already exists
Showtime already exists in DB
Program with unique id 12073476 already exists
Showtime already exists in DB
Program with unique id 1156186051 already exists
Showtime already exists in DB
Program with unique id 14270681 already exists
Showtime already exists in DB
Program with unique id 1156186047 already exists
Showtime already exists in DB
Program with unique id 12185596 already exists
Showtime already exists in DB
Program with unique id 1164453443 already exists
Showtime already exists in DB
Unique Program Petits secrets entre voisins with id 1173541794 added to DB
Showtime: 2018-02-07 00:00:00 on Petits secrets entre voisins was added to DB.
Unique Program My Family with id 327197618 added to DB
Showtime: 2

Program with unique id 5654392724 already exists
Showtime already exists in DB
Unique Program Shortland Street with id 5774958304 added to DB
Showtime: 2018-02-08 00:00:00 on Shortland Street was added to DB.
Program with unique id 11742368 already exists
Showtime already exists in DB
Program with unique id 5726138304 already exists
Showtime already exists in DB
Unique Program Au nom de la vérité with id 5760483598 added to DB
Showtime: 2018-02-08 00:00:00 on Au nom de la vérité was added to DB.
Program with unique id 12127388 already exists
Showtime already exists in DB
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 12744132 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5774822158 added to DB
Showtime: 2018-02-08 00:00:00 on Home And Away was add

Showtime already exists in DB
Unique Program Shortland Street with id 5774958302 added to DB
Showtime: 2018-02-09 00:00:00 on Shortland Street was added to DB.
Program with unique id 5685576507 already exists
Showtime already exists in DB
Program with unique id 4758673325 already exists
Showtime already exists in DB
Unique Program Au nom de la vérité with id 5767194005 added to DB
Showtime: 2018-02-09 00:00:00 on Au nom de la vérité was added to DB.
Program with unique id 12127388 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 12748929 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5774822160 added to DB
Showtime: 2018-02-09 00:00:00 on Home And Away was added to DB.
Program with unique id 13010659 alrea

Unique Program La máscara del Zorro with id 1122343 added to DB
Showtime: 2018-02-10 00:00:00 on La máscara del Zorro was added to DB.
Unique Program Julie Lescaut with id 12062714 added to DB
Showtime: 2018-02-10 00:00:00 on Julie Lescaut was added to DB.
Program with unique id 5787051654 already exists
Showtime already exists in DB
Unique Program Fawlty Towers with id 1267790 added to DB
Showtime: 2018-02-10 00:00:00 on Fawlty Towers was added to DB.
Unique Program Drop Dead Weird with id 5787051642 added to DB
Showtime: 2018-02-10 00:00:00 on Drop Dead Weird was added to DB.
Program with unique id 555109330 already exists
Showtime already exists in DB
Program with unique id 5774958296 already exists
Showtime already exists in DB
Unique Program Navy: investigación criminal with id 17618971 added to DB
Showtime: 2018-02-10 00:00:00 on Navy: investigación criminal was added to DB.
Program with unique id 5774958332 already exists
Showtime already exists in DB
Program with unique id 5630

Program with unique id 4113170753 already exists
Showtime already exists in DB
Program with unique id 5672309340 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 769787178 added to DB
Showtime: 2018-02-12 00:00:00 on Crímenes imperfectos was added to DB.
Program with unique id 5728854777 already exists
Showtime already exists in DB
Unique Program Crímenes imperfectos with id 769787189 added to DB
Showtime: 2018-02-12 00:00:00 on Crímenes imperfectos was added to DB.
Unique Program Las primeras 48 horas with id 5154062965 added to DB
Showtime: 2018-02-12 00:00:00 on Las primeras 48 horas was added to DB.
Program with unique id 5736556671 already exists
Showtime already exists in DB
Program with unique id 5726138304 already exists
Showtime already exists in DB
Unique Program Al Rojo Vivo with id 5788153328 added to DB
Showtime: 2018-02-12 00:00:00 on Al Rojo Vivo was added to DB.
Unique Program The Doctors with id 4587682776 added to DB
Showtime: 2

Unique Program The Doctors with id 4590293384 added to DB
Showtime: 2018-02-13 00:00:00 on The Doctors was added to DB.
Unique Program Al Rojo Vivo with id 5795717431 added to DB
Showtime: 2018-02-13 00:00:00 on Al Rojo Vivo was added to DB.
Program with unique id 5772881933 already exists
Showtime already exists in DB
Program with unique id 5710783827 already exists
Showtime already exists in DB
Unique Program Dr. Phil with id 4566142524 added to DB
Showtime: 2018-02-13 00:00:00 on Dr. Phil was added to DB.
Program with unique id 5714801877 already exists
Showtime already exists in DB
Program with unique id 5795717431 already exists
Showtime already exists in DB
Unique Program Shortland Street with id 5787051750 added to DB
Showtime: 2018-02-13 00:00:00 on Shortland Street was added to DB.
Program with unique id 5751726303 already exists
Showtime already exists in DB
Unique Program The Zoo with id 504402133 added to DB
Showtime: 2018-02-13 00:00:00 on The Zoo was added to DB.
Unique P

Program with unique id 5767194005 already exists
Showtime already exists in DB
Program with unique id 5747865513 already exists
Showtime already exists in DB
Unique Program Shortland Street with id 5787051748 added to DB
Showtime: 2018-02-14 00:00:00 on Shortland Street was added to DB.
Program with unique id 5706150785 already exists
Showtime already exists in DB
Program with unique id 11742368 already exists
Showtime already exists in DB
Unique Program Au nom de la vérité with id 5779591459 added to DB
Showtime: 2018-02-14 00:00:00 on Au nom de la vérité was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 13083441 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5787051702 added to DB
Showtime: 2018-02-14 00:00:00 on Home And Away was a

Program with unique id 12127388 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 1441510 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5787051696 added to DB
Showtime: 2018-02-15 00:00:00 on Home And Away was added to DB.
Program with unique id 12999695 already exists
Showtime already exists in DB
Program with unique id 4077761636 already exists
Showtime already exists in DB
Unique Program Neighbours with id 5770611002 added to DB
Showtime: 2018-02-15 00:00:00 on Neighbours was added to DB.
Program with unique id 3286898281 already exists
Showtime already exists in DB
Unique Program Neighbours with id 5770611000 added to DB
Showtime: 2018-02-15 00:00:00 on Neighbours was added to DB.
Program with unique i

Unique Program Can't Stop Dancing with id 5787051594 added to DB
Showtime already exists in DB
Unique Program Les bracelets rouges with id 5759484904 added to DB
Showtime: 2018-02-16 00:00:00 on Les bracelets rouges was added to DB.
Program with unique id 12147320 already exists
Showtime already exists in DB
Program with unique id 21714748 already exists
Showtime already exists in DB
Unique Program The Late Late Show with id 5804446416 added to DB
Showtime: 2018-02-16 00:00:00 on The Late Late Show was added to DB.
Program with unique id 4107836675 already exists
Showtime already exists in DB
Unique Program Equipo de investigación with id 5421209160 added to DB
Showtime: 2018-02-16 00:00:00 on Equipo de investigación was added to DB.
Program with unique id 13358395 already exists
Showtime already exists in DB
Program with unique id 12108469 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Program with unique id 1887

Showtime: 2018-02-17 00:00:00 on The Walking Dead was added to DB.
Program with unique id 5787053338 already exists
Showtime already exists in DB
Program with unique id 3452333967 already exists
Showtime already exists in DB
Program with unique id 1887442 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Program with unique id 1343917063 already exists
Showtime already exists in DB
Program with unique id 13358395 already exists
Showtime already exists in DB
Unique Program Forget Paris with id 425490 added to DB
Showtime: 2018-02-18 00:00:00 on Forget Paris was added to DB.
Program with unique id 1076761458 already exists
Showtime already exists in DB
Program with unique id 5072179538 already exists
Showtime already exists in DB
Unique Program NCIS with id 1941711788 added to DB
Showtime: 2018-02-18 00:00:00 on NCIS was added to DB.
Program with unique id 11742116 already exists
Showtime already exists in DB
Program w

Program with unique id 5767194005 already exists
Showtime already exists in DB
Unique Program Shortland Street with id 5813225316 added to DB
Showtime: 2018-02-19 00:00:00 on Shortland Street was added to DB.
Unique Program Au nom de la vérité with id 5795613654 added to DB
Showtime: 2018-02-19 00:00:00 on Au nom de la vérité was added to DB.
Unique Program The Zoo with id 522540909 added to DB
Showtime: 2018-02-19 00:00:00 on The Zoo was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Program with unique id 12422039 already exists
Showtime already exists in DB
Program with unique id 11939294 already exists
Showtime already exists in DB
Program with unique id 11909047 already exists
Showtime already exists in DB
Unique Program Home And Away with id 5790692463 added to DB
Showtime: 2018-02-19 00:00:00 on Home And Away was added to DB.
Program with unique id 12830618 already exists
Showtime already exists in DB
Program with unique id 4077761636 a

Program with unique id 3988698367 already exists
Showtime already exists in DB
Unique Program El intermedio with id 5820797235 added to DB
Showtime: 2018-02-20 00:00:00 on El intermedio was added to DB.
Program with unique id 12147320 already exists
Showtime already exists in DB
Program with unique id 5522188903 already exists
Showtime already exists in DB
Unique Program Prime Time with id 5813225264 added to DB
Showtime: 2018-02-20 00:00:00 on Prime Time was added to DB.
Unique Program Sissi : naissance d'une impératrice with id 17897082 added to DB
Showtime: 2018-02-20 00:00:00 on Sissi : naissance d'une impératrice was added to DB.
Unique Program Blue Bloods with id 5421127235 added to DB
Showtime: 2018-02-20 00:00:00 on Blue Bloods was added to DB.
Unique Program In the Dark with id 5033805134 added to DB
Showtime: 2018-02-20 00:00:00 on In the Dark was added to DB.
Unique Program Sissi : naissance d'une impératrice with id 17897085 added to DB
Showtime: 2018-02-20 00:00:00 on Siss

Unique Program Operation Transformation with id 5813225240 added to DB
Showtime already exists in DB
Program with unique id 12746349 already exists
Showtime already exists in DB
Unique Program All Round to Mrs Brown's with id 4771165689 added to DB
Showtime: 2018-02-21 00:00:00 on All Round to Mrs Brown's was added to DB.
Program with unique id 1468795929 already exists
Showtime already exists in DB
Program with unique id 13071731 already exists
Showtime already exists in DB
Program with unique id 1887442 already exists
Showtime already exists in DB
Program with unique id 3452333967 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Program with unique id 13358395 already exists
Showtime already exists in DB
Unique Program Ireland's Fittest Family with id 883030160 added to DB
Showtime: 2018-02-22 00:00:00 on Ireland's Fittest Family was added to DB.
Unique Program Body Shopping with id 4740152057 added to DB
Showtime

Unique Program Le secret du monde englouti with id 12593460 added to DB
Showtime: 2018-02-22 00:00:00 on Le secret du monde englouti was added to DB.
Unique Program School Life with id 5813225272 added to DB
Showtime: 2018-02-22 00:00:00 on School Life was added to DB.
Unique Program Feud: Bette and Joan with id 4571344591 added to DB
Showtime: 2018-02-22 00:00:00 on Feud: Bette and Joan was added to DB.
Program with unique id 13358395 already exists
Showtime already exists in DB
Program with unique id 643777741 already exists
Showtime already exists in DB
Program with unique id 3452333967 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Program with unique id 1887442 already exists
Showtime already exists in DB
Program with unique id 13071731 already exists
Showtime already exists in DB
Unique Program Longmire with id 4099021300 added to DB
Showtime: 2018-02-23 00:00:00 on Longmire was added to DB.
Program with uni

Program with unique id 13358395 already exists
Showtime already exists in DB
Program with unique id 461044 already exists
Showtime already exists in DB
Program with unique id 3452333967 already exists
Showtime already exists in DB
Program with unique id 1887442 already exists
Showtime already exists in DB
Unique Program Equipo de investigación with id 5857876080 added to DB
Showtime: 2018-02-24 00:00:00 on Equipo de investigación was added to DB.
Program with unique id 1076761458 already exists
Showtime already exists in DB
Unique Program Silent Witness with id 4410935026 added to DB
Showtime: 2018-02-24 00:00:00 on Silent Witness was added to DB.
Program with unique id 5072179538 already exists
Showtime already exists in DB
Program with unique id 4617665091 already exists
Showtime already exists in DB
Program with unique id 11742116 already exists
Showtime already exists in DB
Program with unique id 12127388 already exists
Showtime already exists in DB
Program with unique id 581322529

Program with unique id 4098771666 already exists
Showtime already exists in DB
Program with unique id 4120301248 already exists
Showtime already exists in DB
Program with unique id 218925 already exists
Showtime already exists in DB
Program with unique id 21714748 already exists
Showtime already exists in DB
Program with unique id 4098775249 already exists
Showtime already exists in DB
Program with unique id 4528725581 already exists
Showtime already exists in DB
Program with unique id 4098771664 already exists
Showtime already exists in DB
Unique Program Mass with id 5837038271 added to DB
Showtime: 2018-02-25 00:00:00 on Mass was added to DB.
Program with unique id 4120301230 already exists
Showtime already exists in DB
Unique Program Petits secrets entre voisins with id 4142698526 added to DB
Showtime: 2018-02-25 00:00:00 on Petits secrets entre voisins was added to DB.
Program with unique id 11914776 already exists
Showtime already exists in DB
Program with unique id 4142698524 alr

Unique Program Más vale tarde with id 5835149689 added to DB
Showtime: 2018-02-26 00:00:00 on Más vale tarde was added to DB.
Program with unique id 22271793 already exists
Showtime already exists in DB
Program with unique id 19733321 already exists
Showtime already exists in DB
Program with unique id 12766064 already exists
Showtime already exists in DB
Program with unique id 22301334 already exists
Showtime already exists in DB
Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Unique Program Grey's Anatomy with id 22301333 added to DB
Showtime: 2018-02-26 00:00:00 on Grey's Anatomy was added to DB.
Unique Program Nationwide with id 5837038273 added to DB
Showtime: 2018-02-26 00:00:00 on Nationwide was added to DB.
Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program Follow Donal...to Europe with id 3522350440 added to DB
Showtime: 2018-02-26 

Program with unique id 12039424 already exists
Showtime already exists in DB
Program with unique id 11989490 already exists
Showtime already exists in DB
Unique Program Grey's Anatomy with id 22335737 added to DB
Showtime: 2018-02-27 00:00:00 on Grey's Anatomy was added to DB.
Unique Program eco.eye with id 5837038253 added to DB
Showtime: 2018-02-27 00:00:00 on eco.eye was added to DB.
Program with unique id 319923221 already exists
Showtime already exists in DB
Unique Program EastEnders with id 5837433998 added to DB
Showtime: 2018-02-27 00:00:00 on EastEnders was added to DB.
Program with unique id 12044548 already exists
Showtime already exists in DB
Unique Program Guépardes with id 5823541818 added to DB
Showtime: 2018-02-27 00:00:00 on Guépardes was added to DB.
Program with unique id 5599159988 already exists
Showtime already exists in DB
Unique Program Fair City with id 5837038257 added to DB
Showtime: 2018-02-27 00:00:00 on Fair City was added to DB.
Unique Program Sissi impér

Unique Program Marseille with id 5827250903 added to DB
Showtime: 2018-02-28 00:00:00 on Marseille was added to DB.
Unique Program Operation Transformation with id 5837038307 added to DB
Showtime: 2018-02-28 00:00:00 on Operation Transformation was added to DB.
Program with unique id 12579138 already exists
Showtime already exists in DB
Unique Program The Classroom Divide with id 5259959914 added to DB
Showtime: 2018-02-28 00:00:00 on The Classroom Divide was added to DB.
Program with unique id 13121929 already exists
Showtime already exists in DB
Program with unique id 643777741 already exists
Showtime already exists in DB
Program with unique id 13071731 already exists
Showtime already exists in DB
Program with unique id 13071731 already exists
Showtime already exists in DB
Finished adding programs from epg.csv


channel_id  channel_name  channel_country  start_date  start_time  \
0          False         False            False       False       False   
1          False         False            False       False       False   
2          False         False            False       False       False   
3          False         False            False       False       False   
4          False         False            False       False       False   
5          False         False            False       False       False   
6          False         False            False       False       False   
7          False         False            False       False       False   
8          False         False            False       False       False   
9          False         False            False       False       False   
10         False         False            False       False       False   
11         False         False            False       False       False   
12         False         False            False       False       False   
13         False         False            False       False       False   
14         False         False            False       False       False   
15         False         False            False       False       False   
16         False         False            False       False       False   
17         False         False            False       False       False   
18         False         False            False       False       False   
19         False         False            False       False       False   
20         False         False            False       False       False   
21         False         False            False       False       False   
22         False         False            False       False       False   
23         False         False            False       False       False   
24         False         False            False       False       False   
25         False         False            False       False       False   
26         False         False            False       False       False   
27         False         False            False       False       False   
28         False         False            False       False       False   
29         False         False            False       False       False   
...          ...           ...              ...         ...         ...   
4867       False         False            False       False       False   
4868       False         False            False       False       False   
4869       False         False            False       False       False   
4870       False         False            False       False       False   
4871       False         False            False       False       False   
4872       False         False            False       False       False   
4873       False         False            False       False       False   
4874       False         False            False       False       False   
4875       False         False            False       False       False   
4876       False         False            False       False       False   
4877       False         False            False       False       False   
4878       False         False            False       False       False   
4879       False         False            False       False       False   
4880       False         False            False       False       False   
4881       False         False            False       False       False   
4882       False         False            False       False       False   
4883       False         False            False       False       False   
4884       False         False            False       False       False   
4885       False         False            False       False       False   
4886       False         False            False       False       False   
4887       False         False            False       False       False   
4888       False         False 

In order to make this data accessible to the rest of the development teams we need you to
develop an HTTP API consisting of one endpoint that accepts as input:

- start date, end date
- country (optional)
- title (optional)

and returns as output a list of the TV programs with their broadcasting data (channel and
time details) aired within the specified time interval and country, and matching the specified
title.

The list must be in JSON format. For example:

[
{
"program_id": "12044548",
"program_original_title": "Petits plats en équilibre","program_year": null,
"start_time": "20180101 - 13:00",
"end_time": "20180101 - 13:05",
...
},
{
"channel_id": "11939294",
"channel_name": "RTÉ News: One O'Clock",
"program_year": 2001,
"start_time": "20180101 - 13:00",
"end_time": "20180101 - 13:50",
},
...
]

In [ ]:
import urlparse

tv_api_endpoint = 'http://localhost:8080/tv/api/v1/'

params = {
start_date:'YYYY-MM-DD'
end_date='YYYY-MM-DD'
country='Spaìn'
title="Frank's back"}
query_string = '?start_date=YYYY-MM-DD&end_date=YYYY-MM-DD&title=SOME TITLE&country=COUNTRY'

url http://localhost:8080/api/tv/programs?start_date=YYYY-MM-DD&end_date=YYYY-MM-DD&title=SOME

add boolean check to put date parameters in correct order


In [ ]:
params = urllib.unquote(url)

In [ ]:
params.decode('utf-8')

help(urlparse)